src/react.tsx

```diff
@@ -86,10 +90,15 @@ export function useContentfulLiveUpdates<T extends Argument | null | undefined>(
       return;
     }
     // or update content through live updates
-    return ContentfulLivePreview.subscribe(data as Argument, locale, (updatedData) => {
-      // Update the state and adding a version number to it, as some deep nested updates
-      // are not proceeded correctly otherwise
-      update.current((prevState) => ({ data: updatedData as T, version: prevState.version++ }));
+    return ContentfulLivePreview.subscribe({
+      data: data as Argument,
+      // override locale with user locale if passed otherwise use the local from the initialisation
+      locale: locale ? locale : ContentfulLivePreview.locale,
```

Comment:
the locale handling is also in the in the subscibe fn (https://github.com/contentful/live-preview/pull/104/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R135)
could we remove it in the react part?